libs/execution_context/ebpf_program.c

```diff
@@ -2688,3 +2696,40 @@ ebpf_program_set_flags(_Inout_ ebpf_program_t* program, uint64_t flags)
 {
     program->flags = flags;
 }
+
+void
+ebpf_program_set_header_context_descriptor(
+    _In_ const ebpf_context_descriptor_t* context_descriptor, _Inout_ void* program_context)
+{
+    // slot [1] contains the context_descriptor for the program.
+    ebpf_context_header_t* header = CONTAINING_RECORD(program_context, ebpf_context_header_t, context);
+
+    header->context_header[1] = (uint64_t)context_descriptor;
+}
+
+void
+ebpf_program_get_header_context_descriptor(
```

Comment:
same as earlier -- invoked only from `ebpf_program.c`. Should this be marked static?